src/lib/components/Drawer.tsx

```diff
@@ -0,0 +1,69 @@
+import { PropsWithChildren } from 'react';
+import { createPortal } from 'react-dom';
+import styled from 'styled-components';
+import usePortal from '../hooks/usePortal';
+
+type Props = {
+  anchor?: 'left' | 'right' | 'top' | 'bottom';
+  isOpen: boolean;
+  toggleDrawer: () => void;
+};
+
+const Drawer = ({ anchor = 'left', isOpen, toggleDrawer, children }: PropsWithChildren<Props>) => {
+  const { portalRootRef } = usePortal('drawer-root');
+
+  return createPortal(
+    <Backdrop isOpen={isOpen} onClick={toggleDrawer}>
+      <Container isOpen={isOpen} anchor={anchor}>
+        {children}
+      </Container>
+    </Backdrop>,
+    portalRootRef.current,
+  );
+};
+
+export default Drawer;
+
+type BackdropProps = Pick<Props, 'isOpen'>;
+
+type ContainerProps = Omit<Props, 'toggleDrawer'>;
+
+const Backdrop = styled.div<BackdropProps>`
+  visibility: ${props => `${props.isOpen ? 'visible' : 'hidden'}`};
+
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+
+  background-color: ${props => `rgba(0, 0, 0, ${props.isOpen ? 0.5 : 0})`};
+`;
+
+const Container = styled.div<ContainerProps>`
+  display: flex;
+  flex-direction: column;
+  justify-content: ${props => `${props.anchor === 'bottom' ? 'flex-end' : 'flex-start'}`};
+
+  position: fixed;
+  ${props => (props.anchor === 'left' ? 'left: 0;' : '')}
```

Comment:
```suggestion
  ${props => props.anchor === 'left' && css`left: 0`}
```

이렇게도 표현할 수 있을 것 같아요!